### PR TITLE
GUI editor for conditional card

### PR DESCRIPTION
**Pull request by thomasloven**
## Proposed change

image
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                    </div>
+                    <hui-card-editor
+                      .hass=${this.hass}
+                      .value="${this._config.card}"
+                      @config-changed="${this._handleCardChanged}"
+                    ></hui-card-editor>
+                  `
+                : html`
+                    <hui-card-picker
+                      .hass="${this.hass}"
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Nit Picky: This is the only `hass` property you have in quotes :)

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                  .hass=${this.hass}
+                  @change=${this._addCondition}
+                ></ha-entity-picker>
+              </div>
+            </div>
+          `}
+    `;
+  }
+
+  private _selectTab(ev) {
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]:
```suggestion
  private _selectTab(ev): void {
```

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    this._cardTab = parseInt(ev.target.selected, 10) === 1;
+  }
+
+  private _handleCardChanged(ev) {
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]:
```suggestion
  private _handleCardChanged(ev): void {
```

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    ev.stopPropagation();
+    if (!this._config) {
+      return;
+    }
+    this._config.card = ev.detail.config;
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+  private _handleReplaceCard() {
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]:
```suggestion
  private _handleReplaceCard(): void {
```

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+    if (!this._config) {
+      return;
+    }
+    this._config.card = {};
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+
+  private _addCondition(ev) {
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]:
```suggestion
  private _addCondition(ev: MouseEvent): void {
```

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+      return;
+    }
+    this._config.conditions.push({
+      entity: target.value,
+      state: "",
+    });
+    target.value = "";
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+  private _changeCondition(ev) {
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]:
```suggestion
  private _changeCondition(ev: MouseEvent): void {
```

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Quotes should be removed :-)

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts (hunk 2 above)
Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Could be a touch event...

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: `iron-select` is not a `MouseEvent`

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts (hunk 1 above)
Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Thats what I meant :) sorry for the confusion

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+                      <mwc-button @click=${this._handleReplaceCard}
+                        >${this.hass!.localize(
+                          "ui.panel.lovelace.editor.card.conditional.change_type"
+                        )}</mwc-button
+                      >
+                    </div>
+                    <hui-card-editor
+                      .hass=${this.hass}
+                      .value="${this._config.card}"
+                      @config-changed="${this._handleCardChanged}"
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: I don't think these need `""`

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+                    ></hui-card-editor>
+                  `
+                : html`
+                    <hui-card-picker
+                      .hass=${this.hass}
+                      @config-changed="${this._handleCardChanged}"
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: I don't think these need ""

File: src/translations/en.json [hunk 9]
```diff
@@ -1958,7 +1958,14 @@
...
             "conditional": {
               "name": "Conditional",
-              "description": "The Conditional card displays another card based on entity states."
+              "description": "The Conditional card displays another card based on entity states.",
+              "conditions": "Conditions",
+              "card": "Card",
+              "state_equal": "State is equal to",
+              "state_not_equal": "State is not equal to",
+              "current_state": "current",
+              "condition_explanation": "The card will be shown when ALL condition below are fulfilled.",
```
Comment (zsarnett) [inline src/translations/en.json]:
```suggestion
              "condition_explanation": "The card will be shown when ALL conditions below are fulfilled.",
```

File: src/translations/en.json [hunk 10]
```diff
@@ -1958,7 +1958,14 @@
             },
             "conditional": {
               "name": "Conditional",
-              "description": "The Conditional card displays another card based on entity states."
+              "description": "The Conditional card displays another card based on entity states.",
+              "conditions": "Conditions",
+              "card": "Card",
+              "state_equal": "State is equal to",
+              "state_not_equal": "State is not equal to",
+              "current_state": "current",
```
Comment (zsarnett) [inline src/translations/en.json]:
```suggestion
              "current_state": "Current",
```

should this be capitalized?

Comment (thomasloven) [reply src/translations/en.json]: I'm not sure. It's the "current" in `State (current: 'on')`...

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+      if (target.configValue === "entity") {
+        condition.entity = target.value;
+      } else if (target.configValue === "state") {
+        if (condition.state_not !== undefined) {
+          condition.state_not = target.value;
+        } else {
+          condition.state = target.value;
+        }
+      } else if (target.configValue === "invert") {
+        if (parseInt(target.selected, 10) === 1) {
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Is selected a string or why is this needed?
```suggestion
        if (Number(target.selected) === 1) {
```
or
```suggestion
        if (target.selected === "1") {
```

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts (hunk 11 above)
Comment (thomasloven) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: You said it was
Skärmavbild 2020-03-05 kl  22 28 10

I think it isn't, but better safe than sorry in this case, I think.

Comment (thomasloven) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: Never mind, that iron-tabs.

Comment (thomasloven) [reply src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts]: I did a Science. It's a number.

**Comment by thomasloven** [comment]
I'm getting really good at rebasing because of this PR...
